Update build guide for dist binaries + node install changes

Daeraxa opened the pull request:
Included a new section for building Pulsar binaries using `yarn dist`.

Updated the tabs so they all switch together under a single `core-hacking` label.

Removed references to specific Node.js version. Instead there is now a link to the pulsar/.nvmrc file which should always be the "correct" node version to use.
- Advantages of this are that we do not need to update docs when we bump the minimum or recommended node version. It also makes the guide more generic in that it will work on different builds (e.g. if we have an experimental node 18 branch then `18` should be specified in .nvmrc and the build instructions will still work).

Sertonix commented on docs/docs/launch-manual/sections/core-hacking/sections/building-pulsar.md:
```diff
@@ -31,9 +31,9 @@ For OS or distribution specific instructions see below:
-# Install Node16 (using `nvm` - see above) and enable corepack (for `yarn`)
+# Install Node.js (using `nvm` - see above) and enable corepack (for `yarn`)
 
-nvm install 16
```
maybe write `nvm install <version>` instead to make clear that specifying the version is required. The same for the other cases of `nvm install`

Daeraxa replied:
I don't understand, the entire point of .nvmrc and this PR is so that we no longer need to specify the version in the documentation and instead control it entirely from that one file in the pulsar project:
```
 ~/p/pulsar $  nvm install                                                                    
Found '/home/.../pulsar/.nvmrc' with version <16>
Downloading and installing node v16.18.1...
Downloading https://nodejs.org/dist/v16.18.1/node-v16.18.1-linux-x64.tar.xz...
############################################################################################################################ 100.0%
Computing checksum with sha256sum
Checksums matched!
Now using node v16.18.1 (npm v8.19.2)
```

Sertonix replied:
Oh sorry, I didn't know that nvm uses that version automatically.

Daeraxa commented:
I made a mistake in the instructions if somebody was following them exactly. As they were the `node install` wouldn't have used the .nvmrc as they would not have been in the Pulsar dir.
I've moved the `nvm install` out of the platform specific instructions (as it appears all platforms are able to use nvm given the instructions submitted by people) and into the "generic" instructions after the `cd pulsar`. Also added a check with `nvm use` in the next step.

confused-Techie reviewed (dismissed):
As long as the commands here have been tested or confirmed to work on the various OS's the wording and syntax looks great. I'd say lets merge it

Daeraxa commented:
I might need to check the Windows one but the Linux ones definitely work (and macOS would be expected to as well but we haven't done the instructions for it yet). 
Windows I think is a bit weird and we might need to do some extra digging because there are two `nvm`s. The normal "script" nvm works on linux, macos and Windows via some POSIX thing WSL or cygwin but there is another nvm specifically for windows - https://github.com/coreybutler/nvm-windows.

I think you added the windows instructions, what did you use?

confused-Techie commented on docs/docs/launch-manual/sections/core-hacking/sections/building-pulsar.md:
```diff
@@ -10,8 +10,8 @@ The Pulsar application code can be found in the
-- Node.js 16 (recommended installation is via
-  [nvm](https://github.com/nvm-sh/nvm))
+- Node.js (version specified in [pulsar/.nvmrc](https://github.com/pulsar-edit/pulsar/blob/master/.nvmrc),
+  recommended installation is via [nvm](https://github.com/nvm-sh/nvm))
```
```suggestion
  recommended installation is via [nvm](https://github.com/nvm-sh/nvm)) (For Windows User's [nvm-windows](https://github.com/coreybutler/nvm-windows) is recommended.)
```

Daeraxa replied:
`User's` -> `users` but otherwise I think that is good to go.
Do you know if the Windows one fully supports .nvmrc on `nvm install` like the Linux one? The docs were a bit fuzzy, they implied it is supported with `nvm use` but don't explicitly mention it for `install`

confused-Techie replied:
I'll have to go ahead and double check. I'm pretty sure I fully tested the first time making these recommendations, but I'll go ahead and confirm.

confused-Techie reviewed (commented):
Just saw your comment. I've got `nvm-windows` installed, and is what I tested with originally. So I've added that to the docs, and if it looks good to you, lets commit it, and I'll merge.

Daeraxa commented:
@confused-Techie have you had a chance to check the windows nvm bit above?

confused-Techie commented:
@Daeraxa alright, was finally able to confirm this. Sorry for such a long wait.

On Windows 11 I tested and successfully used:

* https://github.com/coreybutler/nvm-windows
* `nvm install`
* `nvm use`
* `corepack enable` (I had to run `nvm use` before I was able to enable, but may have been PowerShell weirdness)

Then with the other dependencies listed I was able to go ahead and build.

Daeraxa commented:
In that case I think with an approval then we are good to go.

confused-Techie reviewed (approved):
> In that case I think with an approval then we are good to go.

Awesome, I'll go ahead and merge this one, glad to see it!
